posts/2020/11/Design In Open Source

```diff
@@ -0,0 +1,39 @@
+At the beginning we mentioned imagining a better world as a part of being a designer, and imagining a healthier open source experience is no exception! Changing the collaboration process is a good first step.
+As a Human-Computer Interaction practitioner, I always strive to understand ‘why’ a problem exists before I decide what to do about it. There’s a lot of ‘what’ and ‘how’ in open-source development. Oftentimes a scientist or developer simply solves the problem they have. But if we take some time to walk through the design process and understand more about why the question arose and who else has the problem, we can get to the core of the problem, and design both a solution and interface for it in a way that is knowledgeable both of the problem and the people solving it. In essence we build software that the user can partner with to solve their problems, instead of building a tool that accomplishes a single task.
+Unfortunately, in practice, there’s been a lot of times we’re asked to give a feature a ‘UX Review’ when what it really needs is a complete design. I’m not suggesting we block all progress on designers, but designers need to be brought into the conversation as soon as possible. This is not to say that the people writing the code are to blame. In order to offer something meaningful to the early design process, designers will have to have a working understanding of the software, including technical constraints. There are two overarching duties in the field of User Experience Design, creating new features (discovering useful features), and refining software (that is already useful). We’re almost never working on refining an existing software tool to work the best it possibly could, that’s usually done by large corporations with matching design budgets. Open-source software is popular, generally speaking, because it’s innovative, not because it has the best user experience. When you’re a designer working in the innovation space you have to be keenly aware of software limitations. 
+This brings us back to a foundational aspect of open-source, mainly, open-source software is built on trust and relationships between contributors. Design in open-source has to be a two-way street. Developers need to be open to digging in and understanding the problem from its source, and designers need to be willing to put in the work to understand the technical limitations of the technology that’s being utilized to build the software. Fortunately this gives designers and developers a place to meet in the middle, I’ve had a number of experiences where simply asking hard questions has helped developers to better understand why they’re solving the problem they are. At the same time, there have been hundreds of solutions I have conceived of that just aren’t possible to implement yet. Getting involved early, and understanding what the limitations are is a great way to help existing teams integrate design into their process.
```

Comment:
Add quirkshop link and footnotes. 
```suggestion
This brings us back to a foundational aspect of open-source, mainly, open-source software is built on trust and relationships between contributors. Design in open-source has to be a two-way street. Developers need to be open to digging in and understanding the problem from its source, and designers need to be willing to put in the work to understand the technical limitations of the technology that’s being utilized to build the software. Fortunately this gives designers and developers a place to meet in the middle, I’ve had a number of experiences where simply asking hard questions has helped developers to better understand why they’re solving the problem they are. At the same time, there have been hundreds of solutions I have conceived of that just aren’t possible to implement yet. Getting involved early, and understanding what the limitations are is a great way to help existing teams integrate design into their process.
Want to hear this talked about more? Tim and I had a conversation with Ani Krishnan and Tony Fast [here](https://www.youtube.com/watch?v=3SBwb8ppz5I).
---
¹ Wondering what’s wrong with this scenario? Yellow and white is a low contrast combination that makes it hard to see the icon. It also fails accessibility contrast standards.
² All caps isn’t recommended because it. Depending how it’s implemented, it can be an accessibility concern, too.
³ A common goal of interface designs it to teach users via interaction patterns. Inconsistency is confusing and can break user trust both short and long term.
```